dlc/src/main/scala/org/bitcoins/dlc/testgen/DLCTestVector.scala

```diff
@@ -165,46 +159,115 @@ object DLCTestVector {
       localChangeSPK: WitnessScriptPubKeyV0,
       remoteExtPrivKey: ExtPrivateKey,
       remoteInput: CurrencyUnit,
-      remoteFundingUtxos: Vector[SegwitV0NativeUTXOSpendingInfoFull],
+      remoteFundingUtxos: Vector[P2WPKHV0SpendingInfo],
       remoteChangeSPK: WitnessScriptPubKeyV0,
       timeouts: DLCTimeouts,
       feeRate: SatoshisPerByte)(
       implicit ec: ExecutionContext): Future[DLCTestVector] = {
     val possibleOutcomes = localPayouts.keySet.toVector
 
-    val dlc = BinaryOutcomeDLCWithSelf(
+    val localFundingInputs = localFundingUtxos.map { info =>
+      (info.outPoint.txIdBE, info.output, info.outPoint.vout)
+    }
+    val remoteFundingInputs = remoteFundingUtxos.map { info =>
+      (info.outPoint.txIdBE, info.output, info.outPoint.vout)
+    }
+
+    val offerDLC = BinaryOutcomeDLCClient(
       outcomeWin = possibleOutcomes.head,
       outcomeLose = possibleOutcomes.last,
       oraclePubKey = oracleKey.publicKey,
       preCommittedR = oracleKValue.publicKey,
-      localExtPrivKey = localExtPrivKey,
-      remoteExtPrivKey = remoteExtPrivKey,
-      localInput = localInput,
+      isInitiator = true,
+      extPrivKey = localExtPrivKey,
+      remoteExtPubKey = remoteExtPrivKey.extPublicKey,
+      input = localInput,
       remoteInput = remoteInput,
-      localFundingUtxos = localFundingUtxos,
-      remoteFundingUtxos = remoteFundingUtxos,
-      localWinPayout = localPayouts(possibleOutcomes.head),
-      localLosePayout = localPayouts(possibleOutcomes.last),
+      fundingUtxos = localFundingUtxos,
+      remoteFundingInputs = remoteFundingInputs,
+      winPayout = localPayouts(possibleOutcomes.head),
+      losePayout = localPayouts(possibleOutcomes.last),
       timeouts = timeouts,
       feeRate = feeRate,
-      localChangeSPK = localChangeSPK,
+      changeSPK = localChangeSPK,
       remoteChangeSPK = remoteChangeSPK,
       network = RegTest
     )
 
+    val acceptDLC = BinaryOutcomeDLCClient(
+      outcomeWin = possibleOutcomes.head,
+      outcomeLose = possibleOutcomes.last,
+      oraclePubKey = oracleKey.publicKey,
+      preCommittedR = oracleKValue.publicKey,
+      isInitiator = false,
+      extPrivKey = remoteExtPrivKey,
+      remoteExtPubKey = localExtPrivKey.extPublicKey,
+      input = remoteInput,
+      remoteInput = localInput,
+      fundingUtxos = remoteFundingUtxos,
+      remoteFundingInputs = localFundingInputs,
+      winPayout = localInput + remoteInput - localPayouts(
+        possibleOutcomes.head),
+      losePayout = localInput + remoteInput - localPayouts(
+        possibleOutcomes.last),
+      timeouts = timeouts,
+      feeRate = feeRate,
+      changeSPK = remoteChangeSPK,
+      remoteChangeSPK = localChangeSPK,
+      network = RegTest
+    )
+
     val outcomeHash = CryptoUtil.sha256(ByteVector(realOutcome.getBytes)).flip
     val oracleSig =
       Schnorr.signWithNonce(outcomeHash.bytes, oracleKey, oracleKValue)
 
+    val offerSigReceiveP =
```

Comment:
important change: this is how I am mocking out interaction in tests across the board